### PR TITLE
added instructiont::get_other()

### PR DESCRIPTION
**Pull request by kroening**
This will eventually enable us to prevent direct access to
instructiont::code.

- [X] Each commit message has a non-empty body, explaining why the change was made.
- [X] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: f7e54866b636de0df1f1d933071c556c0e2cbdb2).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/98701639
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

File: src/analyses/goto_rw.cpp [hunk 1]
```diff
@@ -781,7 +781,7 @@ void goto_rw(
 
   case OTHER:
     // if it's printf, mark the operands as read here
-    if(target->code.get(ID_statement)==ID_printf)
+    if(target->get_other().get_statement() == ID_printf)
     {
       forall_expr(it, target->code.operands())
```
Comment (tautschnig) [inline src/analyses/goto_rw.cpp]: Should also be `target->get_other()`

**Review by tautschnig — APPROVED** [review]

File: src/analyses/goto_rw.cpp (hunk 1 above)
Comment (kroening) [reply src/analyses/goto_rw.cpp]: indeed

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 6a66a514cac03645d9d553fb674eeb30a0b71afa).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/98703991
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by smowton — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: e2c44097dcd2df05e1937ac88d2d9a55967c35ce).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/98715114

**Pull request merged**